azure-kusto-data/azure/kusto/data/_models.py

```diff
@@ -0,0 +1,104 @@
+import json
+from enum import Enum
+from . import _converters
+
+
+class WellKnownDataSet(Enum):
+    """Categorizes data tables according to the role they play in the data set that a Kusto query returns."""
+
+    PrimaryResult = "PrimaryResult"
+    QueryCompletionInformation = "QueryCompletionInformation"
+    TableOfContents = "TableOfContents"
+    QueryProperties = "QueryProperties"
+
+
+class KustoResultRow(object):
+    """Iterator over a Kusto result row."""
+
+    def __init__(self, columns, row):
+        self.columns = columns
+        self.row = row
+
+        # Here we keep converter functions for each type that we need to take special care
+        # (e.g. convert)
+        self.convertion_funcs = {
+            "datetime": _converters.to_datetime,
+            "timespan": _converters.to_timedelta,
+            "DateTime": _converters.to_datetime,
+            "TimeSpan": _converters.to_timedelta,
+        }
+
+    @property
+    def columns_count(self):
+        return len(self.columns)
```

Comment:
Just make sure we do not go thru the whole structure every time here

Reply:
talked about it, o(1)